cg/meta/upload/scout/uploadscoutapi.py

```diff
@@ -246,22 +244,6 @@ def upload_fusion_report_to_scout(
 
         LOG.info("Upload %s fusion report finished!", report_type)
 
-    @staticmethod
-    def _check_cases_single_connected(
-        dna_cases: [models.Family], rna_sample: models.Sample
-    ) -> None:
-        if not dna_cases:
-            raise CgDataError(
-                "Failed to upload RNA sample %s since there are no DNA samples linked to it via subject_id"
-                % rna_sample.internal_id,
-            )
-
-        if len(dna_cases) > 1:
-            raise CgDataError(
-                "Failed to upload RNA sample %s since there are more than one DNA sample linked to it via subject_id"
-                % rna_sample.internal_id,
-            )
-
     def upload_rna_coverage_bigwig_to_scout(self, case_id: str, dry_run: bool) -> None:
```

Comment:
This function is too long and does too much at the moment. I suggest splitting it up into at least one smaller functions. First a function that creates a dict that maps RNA_samples to DNA_samples and their cases. For the structure of the dict I'm thinking something like `{rna_samp1:{dna_samp1:[dna_case1,dna_case2]}, rna_samp2:{dna_samp2:[dna_case2]}}`

Then maybe a second one that loops trough it, but we can decide after the first one is done.

This structure is applicable to the functions that care about samples. For the fusion report a similar but simpler structure is probably sufficient.

Reply:
We now have the create_rna_dna_mapper function, which is the dictionary holding the structure you describe. For now the looping is done by a nested for loop.

Reply:
With the latest commit, we now have a short and simplified function for the fusion report that instead of at sample level, looks at the case level and adds the fusion files.